From 6559c36e7867088d441d24157c463b719b04c25f Mon Sep 17 00:00:00 2001
From: Emmanuel Berthier <emmanuel.berthier@intel.com>
Date: Thu, 12 Nov 2015 10:43:24 +0100
Subject: [PATCH] [MDM] Correct one out-of-bounds on stack issue reported by
 KASAN

Correct one allocation in the stack of a string.

Change-Id: I0ac28648e29d983e739b8be268b22b079827e393
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-8521
Signed-off-by: Emmanuel Berthier <emmanuel.berthier@intel.com>
---
 drivers/staging/modem_control/mdm_ctrl.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/modem_control/mdm_ctrl.c b/drivers/staging/modem_control/mdm_ctrl.c
index 1a86a7d..2509c63 100644
--- a/drivers/staging/modem_control/mdm_ctrl.c
+++ b/drivers/staging/modem_control/mdm_ctrl.c
@@ -789,7 +789,11 @@ static void mdm_cleanup(struct mdm_info *mdm)
 		del_timer(&mdm->flashing_timer);
 		mutex_destroy(&mdm->lock);
 #ifdef CONFIG_HAS_WAKELOCK
-		wake_lock_destroy(&mdm->stay_awake);
+		{ 
+			char *name = (char *)mdm->stay_awake.ws.name;
+			wake_lock_destroy(&mdm->stay_awake);
+			kfree(name);
+		}
 #endif
 	}
 }
@@ -911,7 +915,7 @@ static int mdm_ctrl_module_probe(struct platform_device *pdev)
 
 #ifdef CONFIG_HAS_WAKELOCK
 		snprintf(name, sizeof(name), "%s-wakelock%d", DRVNAME, i);
-		wake_lock_init(&mdm->stay_awake, WAKE_LOCK_SUSPEND, name);
+		wake_lock_init(&mdm->stay_awake, WAKE_LOCK_SUSPEND, kstrdup(name, GFP_KERNEL));
 #endif
 
 		mdm_ctrl_set_state(mdm, MDM_CTRL_STATE_OFF);
-- 
1.9.1

